libs/counterfactuals/src/lib/CounterfactualComponent.tsx

```diff
@@ -186,8 +216,50 @@ export class CounterfactualComponent extends React.PureComponent<
     this.setTemporaryPointToCopyOfDatasetPoint(indexes[0]);
   }
 
+  private setCounterfactualData = async (
+    absoluteIndex: number
+  ): Promise<void> => {
+    this.setState({
+      isCounterfactualsDataLoading: true
+    });
+    const localCounterfactualData = await getLocalCounterfactualsFromSDK(
+      absoluteIndex,
+      this.state.counterfactualsData?.id,
+      this.context.requestLocalCounterfactuals
+    );
+    if (localCounterfactualData.error) {
```

Comment:
Do we have telemetry events for errors/ user actions, or task to track adding logs?

Reply:
thanks, created backlog item to follow upon this once whole feature is completed. I will add logs going forward in feature dev stage.
[Product Backlog Item 2163955](https://msdata.visualstudio.com/Vienna/_workitems/edit/2163955): Add telemetry for large data usage and user clicks